### PR TITLE
Prosthetic surgery now checks if you can attach the bodypart

### LINKED ISSUE #67590: Failing to apply a limb via the prosthetic surgery will get the limb stuck in limbo (your contents)
## Reproduction:

- Remove a monkey's legs. 
- Attempt to replace them with robotic legs.
- You get to the success step with the robotic leg...
- The robotic leg vanishes, and you get a message mentioning their body rejects it. Where'd it go?
   - "...Their body has rejected the robotic right leg!"
- Answer: They are now stuck in your mob contents forever. 
image

### LINKED ISSUE #70125: Prosthetic replacement (head) surgery on HARS'd body fails and puts head into surgeons VV contents
Reporting client version: 514.1588

## Round ID:
[191164](https://scrubby.melonmesa.com/round/191164)

## Reproduction:
Originally found in 191160, retested in Thunderdome on 191164

1- Spawn 2 NPCs, give one HARs
2- Amputate Head from second NPC
3- Perform Prosthetic Replacement surgery on HARS'd corpse with head of second NPC

Surgery fails with a message of "You fail in replacing Unknown's head! Their body has rejected "NPC name"'s head!"

4-  Check your own contents in VV and see that the head is now inside your mob but cannot be found in bag/pockets/etc
image

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

I split can_attach_limb off off attach_limb to allow prosthetic surgery to check if said limb can be put in BEFORE we get to success.

## Why It's Good For The Game

Closes https://github.com/tgstation/tgstation/issues/67590
Closes https://github.com/tgstation/tgstation/issues/70125

## Changelog

:cl:
fix: Prosthetic replacement surgery should no longer move limbs into the player's contents if it fails to attach the limb.
/:cl:

### CODE REVIEW

**Comment by Kapu1178** [comment]
attach_limb() already returned FALSE if you cant attach

**Comment by JohnFulpWillard** [comment]
> attach_limb() already returned FALSE if you cant attach

But it also attaches

File: code/modules/surgery/prosthetic_replacement.dm [hunk 1]
```diff
@@ -53,6 +53,10 @@
 				if(human_target.dna.species.id != bodypart_to_attach.limb_id)
 					organ_rejection_dam = 30
 
+			if(!bodypart_to_attach.can_attach_limb(target))
+				to_chat(user, span_warning("You fail in attaching [target]'s [parse_zone(target_zone)]! Their body has rejected [limb_to_attach]!"))
```
Comment (san7890) [inline code/modules/surgery/prosthetic_replacement.dm:57]:
```suggestion
				to_chat(user, span_warning("You fail in attaching [target]'s [parse_zone(target_zone)]! Their body has rejected [bodypart_to_attach]!"))
```

yo code ain't compiling playa

Comment (JohnFulpWillard) [reply code/modules/surgery/prosthetic_replacement.dm]: ur too late

Comment (san7890) [reply code/modules/surgery/prosthetic_replacement.dm]: nah bruh this was sent 30 seconds before your commit hit remote

**Comment by Kapu1178** [comment]
not a super big fan of the syntax but atleast rename `attach_limb` to `try_attach_limb` to not confuse people i guess

File: code/modules/surgery/bodyparts/dismemberment.dm [hunk 2]
```diff
@@ -473,7 +480,7 @@
 		return FALSE
 	limb = newBodyPart(limb_zone, 0, 0)
 	if(limb)
-		if(!limb.attach_limb(src, 1))
+		if(!limb.try_attach_limb(src, 1))
```
Comment (Time-Green) [inline code/modules/surgery/bodyparts/dismemberment.dm:483]: True instead of 1?

Comment (JohnFulpWillard) [reply code/modules/surgery/bodyparts/dismemberment.dm:483]: the var on whether I did your mom or not is set to 1

Comment (Kapu1178) [reply code/modules/surgery/bodyparts/dismemberment.dm:483]: GOTTEM

File: code/modules/surgery/prosthetic_replacement.dm [hunk 3]
```diff
@@ -53,6 +53,10 @@
 				if(human_target.dna.species.id != bodypart_to_attach.limb_id)
 					organ_rejection_dam = 30
 
+			if(!bodypart_to_attach.can_attach_limb(target))
+				to_chat(user, span_warning("[bodypart_to_attach] can't be attached to [target]'s [parse_zone(target_zone)]!"))
```
Comment (Mothblocks) [inline code/modules/surgery/prosthetic_replacement.dm:57]:
```suggestion
				balloon_alert(user, "that doesn't go on the [parse_zone(target_zone)]!")
```

**Pull request merged**